burst/src/commonMain/kotlin/app/cash/burst/TestInterceptor.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2025 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.burst
+
+interface TestInterceptor {
+  fun intercept(test: Test)
+
+  class Test(
```

Comment:
I'm a bit wary of colliding with `@Test` even when nested. Could be `TestFunction` and top-level? I don't really care all that much though.

Reply:
Yeah, I was a bit worried about that. Fixed!